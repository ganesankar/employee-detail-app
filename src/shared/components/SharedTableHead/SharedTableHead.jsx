import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import PropTypes from "prop-types";
import React from "react";
import {makeStyles, withStyles} from '@material-ui/core';
import TableCell from "@material-ui/core/TableCell";
import TableSortLabel from "@material-ui/core/TableSortLabel";

const StyledTableCell = withStyles(theme => ({
  head: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white,
    paddingLeft: 0,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);

const StyledTableSortLabel = withStyles(theme => ({
  root: {
    color: '#fff'
  },
  icon: {
    color: '#fff !important'
  }
}))(TableSortLabel);

const useStyles = makeStyles(theme => ({
  visuallyHidden: {
    border: 0,
    clip: 'rect(0 0 0 0)',
    height: 1,
    margin: -1,
    overflow: 'hidden',
    padding: 0,
    position: 'absolute',
    top: 20,
    width: 1
  },
  action: {
    width: 80,
    paddingLeft: 12
  },
  sortLabel: {
    color: '#fff !important'
  }
}));

const SharedTableHead = (props) => {
  const classes = useStyles();
  const {order, orderBy, onRequestSort} = props;
  const createSortHandler = property => event => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        <StyledTableCell className={classes.action}>Action</StyledTableCell>
        {props.headCells.map(headCell => (
          <StyledTableCell
            key={headCell.key}
            sortDirection={orderBy === headCell.key ? order : false}>
            <StyledTableSortLabel
              className={classes.sortLabel}
              active={orderBy === headCell.key}
              direction={order}
              onClick={createSortHandler(headCell.key)}>
              {headCell.label}
              {orderBy === headCell.key ? (
                <span className={classes.visuallyHidden}>
                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                </span>
              ) : null}
            </StyledTableSortLabel>
          </StyledTableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

SharedTableHead.propTypes = {
  onRequestSort: PropTypes.func.isRequired,
  order: PropTypes.oneOf(['asc', 'desc']).isRequired,
  orderBy: PropTypes.string.isRequired,
  headCells: PropTypes.array.isRequired
};

export default SharedTableHead;
