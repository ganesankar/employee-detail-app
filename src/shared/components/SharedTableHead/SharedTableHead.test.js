import React from 'react';
import SharedTableHead from './SharedTableHead';
import renderer from 'react-test-renderer';

const employees = [
  {
    id: '5d6f70562db7ce28a19c048c',
    Title: 'Ms',
    FirstName: 'Selena',
    LastName: 'Merrill',
    Company: 'KINDALOO'
  },
  {
    id: '5d6f7056adc94ae5fbcfd2da',
    Title: 'Dr',
    FirstName: 'Mckay',
    LastName: 'Wilson',
    Company: 'IPLAX'
  }
];

const headerCells = [
  {label: 'First Name', key: 'FirstName'},
  {label: 'Last Name', key: 'LastName'},
  {label: 'Title', key: 'Title'},
  {label: 'Company', key: 'Company'}
];

const handleRequestSort = () => {
  console.log('on handle sort')
};

it('deep snapshot renders correctly', () => {
  const tree = renderer.create(<SharedTableHead order="desc"
                                                orderBy="FirstName"
                                                headCells={headerCells}
                                                onRequestSort={handleRequestSort}
                                                rowCount={employees.length}/>).toJSON();
  expect(tree).toMatchSnapshot();
});