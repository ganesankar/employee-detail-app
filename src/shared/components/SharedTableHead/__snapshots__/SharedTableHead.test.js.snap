// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot renders correctly 1`] = `
<thead
  className="MuiTableHead-root"
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root makeStyles-action-2 MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-10"
      scope="col"
    >
      Action
    </th>
    <th
      aria-sort="descending"
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-10"
      scope="col"
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-24 makeStyles-sortLabel-3 MuiTableSortLabel-active"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        First Name
        <span
          className="makeStyles-visuallyHidden-1"
        >
          sorted descending
        </span>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-25 MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-10"
      scope="col"
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-24 makeStyles-sortLabel-3"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Last Name
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-25 MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-10"
      scope="col"
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-24 makeStyles-sortLabel-3"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Title
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-25 MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-10"
      scope="col"
    >
      <span
        className="MuiButtonBase-root MuiTableSortLabel-root WithStyles(ForwardRef(TableSortLabel))-root-24 makeStyles-sortLabel-3"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Company
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon WithStyles(ForwardRef(TableSortLabel))-icon-25 MuiTableSortLabel-iconDirectionDesc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
  </tr>
</thead>
`;
